server.go

```diff
@@ -673,7 +1491,28 @@ func (s *server) peerHandler() {
 		permanentPeers = cfg.AddPeers
 	}
 	for _, addr := range permanentPeers {
-		s.handleAddPeerMsg(state, newOutboundPeer(s, addr, true, 0))
+		peerCfg := &peer.Config{
```

Comment:
Same thing here regarding defining it after the connection is established.  Also, this code looks identical to the above code, so it should probably be refactored.